src/lib/storage/reference_segment.cpp

```diff
@@ -0,0 +1,34 @@
+#include <memory>
+
+#include "reference_segment.hpp"
+
+namespace opossum {
+
+ReferenceSegment::ReferenceSegment(const std::shared_ptr<const Table> referenced_table,
+                                   const opossum::ColumnID referenced_column_id,
+                                   const std::shared_ptr<const PosList> pos)
+    : _referenced_table(referenced_table), _referenced_column_id(referenced_column_id), _positions(pos) {}
+
+AllTypeVariant ReferenceSegment::operator[](const ChunkOffset chunk_offset) const {
+  // Get the correct position within the PositionList for the given chunk_offset
```

Comment:
We were reminded in our review that there should be a performance warning.
 `PerformanceWarning("operator[] used");`